server/src/main/java/io/crate/metadata/ColumnIdent.java

```diff
@@ -196,6 +197,21 @@ public boolean isChildOf(ColumnIdent parent) {
         return false;
     }
 
+    /**
+     * Replaces the parent name of this column specified by the 'newPath' e.g.::
+     *   'a.b.c'.replacePath('a.x') -> 'a.x.c' : parent 'a.b' renamed to 'a.x' such that its child 'a.b.c' is renamed to 'a.x.c'
+     *   'a.b.c'.replacePath('x') -> 'x.b.c' : parent 'a' renamed to 'x' such that its child 'a.b.c' is renamed to 'x.b.c'
+     * This method also renames self e.g.::
+     *   'a'.replacePath('b') -> 'b'
+     */
+    @NotNull
+    public ColumnIdent replacePath(@NotNull ColumnIdent newPath) {
```

Comment:
Not sure about this - but maybe `replaceHead` or `replacePrefix` would make this clearer?
It's replacing the starting path segment with the new prefix, and keeps any extra tail it had.